### PR TITLE
ASoC: Intel: Make glk+rt5682 echo ref dynamic

### LINKED ISSUE #1793: [BUG][GLK] glk_rt5682_max98357a in probe deferral loop
Kernel SHA b2d982370860640838a30fd6927eab57b7de615b
SOF SHA: 01a589581abfc786964e67c00ae7e4580b6a6a29

Running latest kernel on glk chromebook with rt5682 we see this loop running because we are missing a CPU DAI but I can't spot anything particular in the machine driver link that has changed, maybe there was a core change that broke things? Any ideas? This is a blocker.

```
[   30.809389] glk_rt5682_max98357a glk_rt5682_max98357a: info: override BE DAI link Glk Audio Echo Reference cap
[   30.809403] glk_rt5682_max98357a glk_rt5682_max98357a: info: override BE DAI link SSP1-Codec
[   30.809411] glk_rt5682_max98357a glk_rt5682_max98357a: info: override BE DAI link SSP2-Codec
[   30.809419] glk_rt5682_max98357a glk_rt5682_max98357a: info: override BE DAI link dmic01
[   30.809428] glk_rt5682_max98357a glk_rt5682_max98357a: info: override BE DAI link iDisp1
[   30.809435] glk_rt5682_max98357a glk_rt5682_max98357a: info: override BE DAI link iDisp2
[   30.809443] glk_rt5682_max98357a glk_rt5682_max98357a: info: override BE DAI link iDisp3                                                  
[   30.809677] glk_rt5682_max98357a glk_rt5682_max98357a: ASoC: CPU DAI Echoref Pin not registered
[   30.825873] glk_rt5682_max98357a glk_rt5682_max98357a: info: override BE DAI link Glk Audio Echo Reference cap
[   30.825890] glk_rt5682_max98357a glk_rt5682_max98357a: info: override BE DAI link SSP1-Codec
[   30.825898] glk_rt5682_max98357a glk_rt5682_max98357a: info: override BE DAI link SSP2-Codec
[   30.825906] glk_rt5682_max98357a glk_rt5682_max98357a: info: override BE DAI link dmic01
[   30.825914] glk_rt5682_max98357a glk_rt5682_max98357a: info: override BE DAI link iDisp1
[   30.825922] glk_rt5682_max98357a glk_rt5682_max98357a: info: override BE DAI link iDisp2
[   30.825930] glk_rt5682_max98357a glk_rt5682_max98357a: info: override BE DAI link iDisp3                                                                                                                        
[   30.826145] glk_rt5682_max98357a glk_rt5682_max98357a: ASoC: CPU DAI Echoref Pin not registered
```

### PR DESCRIPTION
**Pull request by cujomalainey**
Fixes #1793

### CODE REVIEW

**Review by ranj063 — DISMISSED** [review]

File: sound/soc/intel/boards/glk_rt5682_max98357a.c [hunk 1]
```diff
@@ -409,6 +409,7 @@ static struct snd_soc_dai_link geminilake_dais[] = {
 		.init = NULL,
 		.capture_only = 1,
 		.nonatomic = 1,
+		.dynamic = 1,
```
Comment (plbossart) [inline sound/soc/intel/boards/glk_rt5682_max98357a.c:412]: The commit message is a bit misleading. It's more like this is required for SOF to ignore hard-coded front-ends signalled with the .dynamic field

**Review by plbossart — CHANGES_REQUESTED** [review]

**Review by ranj063 — APPROVED** [review]

**Comment by kv2019i** [comment]
SOFCI TEST

**Review by kv2019i — APPROVED** [review]
Thanks, looks good! I'll try to kick the CI again to get at least one set of test results before merging.

**Comment by kv2019i** [comment]
OK, tests are good as well. I'll give @plbossart a chance to ack his comments, but then this is ready for merge.

**Review by plbossart — APPROVED** [review]

**Pull request merged**